src/tests/tribol_proximity_check.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#include <iostream>
+#include <set>
+
+#include <gtest/gtest.h>
+
+// Tribol includes
+#include "tribol/config.hpp"
+#include "tribol/common/Parameters.hpp"
+#include "tribol/interface/tribol.hpp"
+#include "tribol/interface/mfem_tribol.hpp"
+
+#include "shared/mesh/MeshBuilder.hpp"
+
+#ifdef TRIBOL_USE_UMPIRE
+// Umpire includes
+#include "umpire/ResourceManager.hpp"
+#endif
+
+// MFEM includes
+#include "mfem.hpp"
+
+/**
+ * @brief Tests the binning proximity scale parameter for various common Tribol contact setups.
+ *
+ */
+class ProximityTest : public testing::TestWithParam<std::tuple<int, tribol::RealT, tribol::RealT, bool>> {
+ protected:
+  /**
+   * @brief Maximum force computed over the mesh.
+   */
+  double max_force_;
+
+  /**
+   * @brief Binning methods to test for each contact problem.
+   */
+  std::array<tribol::BinningMethod, 3> binning_methods_{ tribol::BINNING_CARTESIAN_PRODUCT, tribol::BINNING_GRID,
+                                                         tribol::BINNING_BVH };
+
+  void UpdateTribol( shared::ParMeshBuilder& mesh, int coupling_scheme_id )
+  {
+    tribol::updateMfemParallelDecomposition();
+    constexpr int cycle = 0;
+    constexpr tribol::RealT t = 0.0;
+    tribol::RealT dt = 1.0;
+    tribol::update( cycle, t, dt );
+
+    mfem::LinearForm r( &mesh.getNodesFESpace() );
+    r = 0.0;
+    tribol::getMfemResponse( coupling_scheme_id, r );
+    max_force_ = r.Max();
+
+    r.Print();
+
+    MPI_Allreduce( MPI_IN_PLACE, &max_force_, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD );
+  }
+
+  void SetUpCommonPlaneProblem( shared::ParMeshBuilder& mesh, const std::set<int>& contact_surf_1,
+                                const std::set<int>& contact_surf_2, tribol::RealT penalty,
+                                tribol::RealT binning_proximity, tribol::BinningMethod binning_method )
+  {
+    // set up tribol
+    constexpr int coupling_scheme_id = 0;
+    constexpr int mesh1_id = 0;
+    constexpr int mesh2_id = 1;
+    tribol::registerMfemCouplingScheme( coupling_scheme_id, mesh1_id, mesh2_id, mesh, mesh.getNodes(), contact_surf_1,
+                                        contact_surf_2, tribol::SURFACE_TO_SURFACE, tribol::NO_CASE,
+                                        tribol::COMMON_PLANE, tribol::FRICTIONLESS, tribol::PENALTY, binning_method );
+    tribol::setMfemKinematicConstantPenalty( coupling_scheme_id, penalty, penalty );
+    tribol::setBinningProximityScale( coupling_scheme_id, binning_proximity );
+
+    UpdateTribol( mesh, coupling_scheme_id );
+  }
+
+  void SetUp2DCommonPlaneProblem( tribol::BinningMethod binning_method )
```

Comment:
Minor: Setup is usually one word so "U" would be lower case :)

Reply:
It's being used as a verb here, so "set up" is correct.  See e.g. https://grammarist.com/spelling/set-up-vs-setup/